### PR TITLE
[BYOC][Verilator] Refactor Verilator runtime

### PR DESCRIPTION
**Pull request by vegaluisjose**
This PR refactors the Verilator runtime to support multiple options, including enabling a profiler for hardware designs that support performance counters. Also, I added some documentation to header files.

After this gets merged, my next goal is to update the Verilator app in VTA to showcase how this profiling information can be used when running actual models.

@tmoreau89 @liangfu

### CODE REVIEW

File: src/runtime/contrib/verilator/verilator_runtime.cc [hunk 1]
```diff
@@ -76,19 +76,24 @@ VerilatorProfiler* VerilatorProfiler::ThreadLocal() {
...
 }
 
-void VerilatorRuntime::LoadLibrary(const std::string& lib_name) {
-  lib_ = new VerilatorLibrary();
-  lib_->Load(lib_name);
+VerilatorRuntime::~VerilatorRuntime() {
+  auto dealloc = reinterpret_cast<VerilatorDeallocFunc>(lib_->GetSymbol("VerilatorDealloc"));
+  ICHECK(dealloc != nullptr);
+  dealloc(device_);
+  lib_->~VerilatorLibrary();
```
Comment (liangfu) [inline src/runtime/contrib/verilator/verilator_runtime.cc:83]: just out of curiosity, why do we explicitly call the deallocator instead of `delete lib_`?

**Review by liangfu — APPROVED** [review]
Thanks @vegaluisjose for bringing the change, especially in enabling the profiler feature. The implementation looks good.

Free feel to merge when you think this is ready to merge. @vegaluisjose

**Review by tmoreau89 — APPROVED** [review]
Thanks LGTM

**Comment by tmoreau89** [comment]
Thank you @vegaluisjose @liangfu , the PR has been merged.

**Pull request merged**